### PR TITLE
Rollup changes

### PR DESCRIPTION
**Pull request by ctubbsii**
Fixes several issues, and cleans up the generation of the source and binary tarballs for releases.

### CODE REVIEW

File: modules/accumulo/src/test/java/org/apache/fluo/accumulo/iterators/TestData.java [hunk 1]
```diff
@@ -158,4 +159,9 @@ public String toString() {
     sb.append("}");
     return sb.toString();
   }
+
+  @Override
+  public int hashCode() {
+    return Objects.hashCode(data);
```
Comment (keith-turner) [inline modules/accumulo/src/test/java/org/apache/fluo/accumulo/iterators/TestData.java]: I did not know of `Objects.hashCode()`

File: modules/api/src/main/java/org/apache/fluo/api/data/Span.java [hunk 2]
```diff
@@ -194,6 +194,11 @@ public String toString() {
         + ((endInclusive && !end.equals(RowColumn.EMPTY)) ? "]" : ")");
   }
 
+  @Override
+  public int hashCode() {
+    return Objects.hash(start, end, startInclusive, endInclusive);
```
Comment (keith-turner) [inline modules/api/src/main/java/org/apache/fluo/api/data/Span.java]: This method is also new to me.  We can use these methods in other places.  Opened #740

File: modules/distribution/pom.xml [hunk 3]
```diff
@@ -31,26 +31,32 @@
     <dependency>
       <groupId>org.apache.fluo</groupId>
       <artifactId>fluo-accumulo</artifactId>
+      <optional>true</optional>
```
Comment (keith-turner) [inline modules/distribution/pom.xml]: Why make these optional?

File: modules/distribution/src/main/assembly/bin.xml [hunk 4]
```diff
@@ -23,25 +23,18 @@
...
+      <useProjectArtifact>false</useProjectArtifact>
       <useTransitiveDependencies>false</useTransitiveDependencies>
       <includes>
-        <include>org.apache.fluo:fluo-api</include>
-        <include>org.apache.fluo:fluo-accumulo</include>
-        <include>org.apache.fluo:fluo-core</include>
-        <include>org.apache.fluo:fluo-mini</include>
-        <include>org.apache.fluo:fluo-mapreduce</include>
-        <include>org.apache.fluo:fluo-cluster</include>
+        <include>${groupId}:fluo-*</include>
```
Comment (keith-turner) [inline modules/distribution/src/main/assembly/bin.xml]: nice

File: modules/mini/src/main/java/org/apache/fluo/mini/MiniFluoImpl.java [hunk 5]
```diff
@@ -214,7 +214,7 @@ public static void main(String[] args) {
         System.exit(-1);
       }
       try (MiniFluo mini = new MiniFluoImpl(config)) {
-        log.info("MiniFluo is running");
+        log.info("MiniFluo is running " + mini);
```
Comment (keith-turner) [inline modules/mini/src/main/java/org/apache/fluo/mini/MiniFluoImpl.java]: Why did you add this to the log message?  I don't think MiniFluoImpl has a `toString()` method.  I am not sure anything calls this main method since the mini-fluo script was removed in  6fd8629.  We may be able to remove it.

**Comment by keith-turner** [comment]
I built this branch and it built fine.   I also looked at the tarball produced and it looked good, but I did not try running fluo from the tarball.

File: pom.xml [hunk 6]
```diff
@@ -59,6 +59,7 @@
     <findbugs.maxRank>9</findbugs.maxRank>
     <hadoop.version>2.6.3</hadoop.version>
     <logback.version>1.1.3</logback.version>
+    <maven.test.redirectTestOutputToFile>true</maven.test.redirectTestOutputToFile>
```
Comment (mikewalch) [inline pom.xml]: This will redirect all logs to a file.  Correct?  I like making the tests less spammy.  However, I don't think it should be at the expense of not seeing error messages in console.  We could instead convert any `System.out.println` call to `log.debug`.  Test logging is configured to not print out any messags below ERROR.

File: modules/mini/src/main/java/org/apache/fluo/mini/MiniFluoImpl.java (hunk 5 above)
Comment (keith-turner) [inline modules/mini/src/main/java/org/apache/fluo/mini/MiniFluoImpl.java]: I looked for Java code and scripts that migh call the main method and did not find any.

File: pom.xml (hunk 6 above)
Comment (keith-turner) [inline pom.xml]: It sounds like it may only redirect stdout to a file.  If the logging goes to stderr, it would not be redirected.  Personally I would rather fix the test than make this change.  When debugging issues I sometimes add temporary println to impl and test code and its nice to easily see it when running the test.

http://maven.apache.org/surefire/maven-surefire-plugin/test-mojo.html#redirectTestOutputToFile

Comment (ctubbsii) [inline pom.xml]: I think it is just STDOUT, not STDERR, like @keith-turner said.
I think converting to the log is also appropriate, but I also think doing this is useful, too. It helps separate the test output from the maven plugin output, so you can distinguish between the two. You can still use temporary print statements to debug, but with this, you'll be able to find that output in a file, separate from the plugin output.

File: modules/distribution/pom.xml (hunk 3 above)
Comment (ctubbsii) [inline modules/distribution/pom.xml]: Because if anybody had a dependency on the tar.gz artifact from Maven, they won't have transitive dependencies on the tarball's dependencies.

File: modules/mini/src/main/java/org/apache/fluo/mini/MiniFluoImpl.java (hunk 5 above)
Comment (ctubbsii) [inline modules/mini/src/main/java/org/apache/fluo/mini/MiniFluoImpl.java]: The toString may not be meaningful at the moment, but it did resolve the compiler warning about an unreferenced resource inside a try-catch block. I initially tried to make it `while (mini != null)`, but findbugs didn't like that... it detected it as a possible unintentional infinite loop (vs. the current code, which is clearly an intentional infinite loop).

I can update the PR to remove the main method if it's not needed at all.

**Comment by keith-turner** [comment]
+1

File: pom.xml (hunk 6 above)
Comment (mikewalch) [inline pom.xml]: It looks like `log.error` messages go to the log.  How about we remove this?  I'll create a follow on ticket and I will remove convert any `System.out.println` to `log.debug` in tests.

**Comment by mikewalch** [comment]
+1

File: pom.xml (hunk 6 above)
Comment (ctubbsii) [inline pom.xml]: I'll strip out that part of this PR before merging, pending that future ticket (#727)
